Metalama.Framework.Engine/Linking/Inlining/ConstructorInliner.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Engine.CodeModel;
+using Metalama.Framework.Engine.Transformations;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Linq;
+
+namespace Metalama.Framework.Engine.Linking.Inlining;
+
+/// <summary>
+/// Constructor inliner.
+/// </summary>
+internal sealed class ConstructorInliner : Inliner
+{
+    private static bool IsInlineableObjectCreation(
+        SemanticModel semanticModel,
+        IMethodSymbol contextConstructor,
+        ObjectCreationExpressionSyntax objectCreationExpression )
+    {
+        var expectedNumberOfParameters =
+            contextConstructor.Parameters.LastOrDefault()?.Name == AspectReferenceSyntaxProvider.LinkerOverrideParamName
+            ? contextConstructor.Parameters.Length - 1
+            : contextConstructor.Parameters.Length;
+
+        return
+            expectedNumberOfParameters == (objectCreationExpression.ArgumentList?.Arguments.Count ?? 0)
+            && (objectCreationExpression.ArgumentList?.Arguments
+               .Select( ( x, i ) => (Argument: x.Expression, Index: i) )
+               ?.All( a => SymbolEqualityComparer.Default.Equals( semanticModel.GetSymbolInfo( a.Argument ).Symbol, contextConstructor.Parameters[a.Index] ) )
+              ?? false);
+    }
+
+    public override bool IsValidForTargetSymbol( ISymbol symbol )
+        => symbol is IMethodSymbol { MethodKind: MethodKind.Constructor };
```

Comment:
~~Should add `IsStatic: false` here and probably in the next line.~~

Actually, this is unnecessary, static constructors have a separate `MethodKind.StaticConstructor`.

Reply:
Yep, I'm going to create a bug for that.